core-bundle/src/Resources/contao/modules/ModuleSearch.php

```diff
@@ -313,13 +315,60 @@ protected function compile()
 					$objTemplate->hasContext = true;
 				}
 
+				$this->addImageToTemplateFromSearchResult($arrResult[$i], $objTemplate);
+
 				$this->Template->results .= $objTemplate->parse();
 			}
 
 			$this->Template->header = vsprintf($GLOBALS['TL_LANG']['MSC']['sResults'], array($from, $to, $count, $strKeywords));
 			$this->Template->duration = System::getFormattedNumber($query_endtime - $query_starttime, 3) . ' ' . $GLOBALS['TL_LANG']['MSC']['seconds'];
 		}
 	}
+
+	protected function addImageToTemplateFromSearchResult(array $result, Template $template): void
+	{
+		$template->hasImage = false;
+
+		if (!isset($result['meta']) || null == $result['meta'])
+		{
+			return;
+		}
+
+		$meta = json_decode($result['meta'], true);
+
+		foreach ($meta as $v)
+		{
+			if (!isset($v['https://schema.org/primaryImageOfPage']))
```

Comment:
```suggestion
			if (!isset($v['https://schema.org/primaryImageOfPage']['contentUrl']))
```
So that we cannot get an undefined array key error in line 347.